mantidproject/mantid
# "Sample logs" dialog - fix filtering options

**Do not merge until #17905 has been merged first**

---

This PR continues on from the first half of this work in #17905, which fixed the statistics and time-averaging on `TimeSeriesProperties` to respect filtering. The current PR extends this to the "Sample logs" dialog, which has radio buttons for filtering by status (`running` log), period or both. 

It also refactors the code that performs this filtering out of `MantidUI` and into a separate class, where tests have been added.

**To test:**

- Verify automated tests cover the functionality of `LogFilterGenerator`
- Load the file `EMU00066122.nxs` at `\\olympic\babylon5\Scratch\TomPerkins\test_tsp_filtering`. This run was aborted, restarted and has several pauses, so is a good test of this work.
- Right-click on the workspace in MantidPlot and select "Sample logs..."
- Select the log "Slits", which was set to 12 at first, then the run was aborted and restarted with "Slits" set to 5. Change the radio button selection (below) and verify that the statistics change with it.
![logs](https://cloud.githubusercontent.com/assets/15363125/20303467/690eb5e4-ab23-11e6-840a-39a4f9694420.PNG)


Fixes #17877.

[Release notes](https://github.com/mantidproject/mantid/blob/78eeb5c43b422de7b6db3e91d5e433f407a55081/docs/source/release/v3.9.0/ui.rst#bugs-resolved) have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
